js/tabs/schedule/EditCourse.js

```diff
@@ -66,7 +66,7 @@ export default function EditCourse() {
           defaultValue={course}
           maxLength={15}
           onChangeText={course =>
-            setCourse(course && course.trim().toUpperCase())
```

Comment:
Dies ist kein Fehler. Mit `course && ...` wird tatsächlich der Fall vermieden, dass `course` `null` oder `undefined` ist.

Reply:
Ja, aber: Ich vermute dass diese Schreibweise mit einem nicht immer reproduzierbaren Bug zusammenhängt. Wenn ein Kurs eingegeben wird kommt es in bestimmten Umständen dazu dass die Eingabe sehr seltsam aussieht. Buchstaben sind teilweise doppelt. Ohne diesen Ausdruck konnten wir den Fehler nicht mehr reproduzieren.

Meine Vermutung dazu wäre, korrigieren Sie mich bitte wenn ich falsch liegen sollte: Wenn der course undefined bzw. null ist, dann wird der Course-State auf false gesetzt? Wäre nicht hier das setzen von null oder "" sinnvoller? Ich würde hier nämlich eine schlechte Reaktion vermuten wenn das defaultValue auf false gesetzt ist. Das würde auch das sporadische Auftreten des Fehlers erklären, da nicht immer Course null ist und somit auch das defaultValue nicht immer auf false gesetzt wird.